### PR TITLE
Articles category module new filter option (modified later than create)

### PR DESCRIPTION
**Pull request by yild**
Pull Request for Issue # .

### Summary of Changes

New additional filter option that can hide articles modified later than provided value of minutes after create date/time.
This is useful for articles list with Date Range Field set to Modified Date - atm we can see all modified articles even when modified=created. My PR will allow to hide articles modified within i.e. 20 mins from creating date/time.
I.E we have two lists on a web page - recently created articles and recently modified. Without my code recently modified will show all articles even if modified value is null (newly created). There are times that users create article and after posting them they make a simple modification (i.e. spelling) and we don't want to list those articles as 'modified'. 

### Testing Instructions

Add new module of type Articles-Category. Set Modified Later Than value to 0 (default), Date Range Field to "modified date".

Create a few articles, after a while (a few mins. i.e. 5) modify some of them. 
Go to web page - all articles should be visible.
Modify Set Modified Later value to 4 (mins). 
Go to web page - all articles that was modified within those 4 mins after creation date/time should be invisible, only those modified > 4 min should be on list.

### Expected result

Check Testing Instructions

### Documentation Changes Required

Description of additional filter options.

### CODE REVIEW

File: components/com_content/models/articles.php [hunk 1]
```diff
@@ -549,6 +549,12 @@ protected function getListQuery()
 					. ' AND ' . $db->quoteName('tagmap.type_alias') . ' = ' . $db->quote('com_content.article')
 				);
 		}
+		
+		// Filter modified later than n minutes from creation
+		if ($this->getState('filter.modified_later_than'))
+		{
+			$query->where('(a.modified > (a.created + interval ' . $this->getState('filter.modified_later_than', 0) . ' minute))');
```
Comment (bertmert) [inline components/com_content/models/articles.php]: Just cosmetic I think: `interval` and `minute` should be `INTERVAL` and `MINUTE` (upper-case).

**Comment by infograf768** [comment]
should you not check that the value entered is an integer?

File: language/en-GB/en-GB.mod_articles_category.ini [hunk 2]
```diff
@@ -50,6 +50,8 @@ MOD_ARTICLES_CATEGORY_FIELD_LINKTITLES_LABEL="Linked Titles"
 MOD_ARTICLES_CATEGORY_FIELD_LINKTITLES_DESC="Linked titles."
 MOD_ARTICLES_CATEGORY_FIELD_MODE_DESC="Please select the mode you would like to use. If Normal Mode is chosen, then simply configure the module and it will display a static list of Articles on the menu items you assign the module to. If Dynamic Mode is chosen, then you can still configure the module normally, however now the Category option will no longer be used. Instead, the module will dynamically detect if you are on a Category view and will display the list of articles within that Category accordingly. When Dynamic Mode is chosen, it is best to leave the module set to display on all pages, as it will decide to display anything dynamically."
 MOD_ARTICLES_CATEGORY_FIELD_MODE_LABEL="Mode"
+MOD_ARTICLES_CATEGORY_FIELD_MODIFIEDLATERTHAN_LABEL="Modified Later Than"
+MOD_ARTICLES_CATEGORY_FIELD_MODIFIEDLATERTHAN_DESC="Only modified later than given value (in minutes) from creation date/time. This option is usefull if Date Range Field is set to Modified Date."
```
Comment (brianteeman) [inline language/en-GB/en-GB.mod_articles_category.ini]: useful has online one l

File: modules/mod_articles_category/mod_articles_category.xml [hunk 3]
```diff
@@ -272,6 +272,14 @@
 					default="30"
 				/>
 
+				<field
+					name="modified_later_than"
+					type="text"
+					label="MOD_ARTICLES_CATEGORY_FIELD_MODIFIEDLATERTHAN_LABEL"
+					description="MOD_ARTICLES_CATEGORY_FIELD_MODIFIEDLATERTHAN_DESC"
+					default="0"
```
Comment (infograf768) [inline modules/mod_articles_category/mod_articles_category.xml]: Add I guess
filter="integer"

**Comment by yild** [comment]
@infograf768 @brianteeman - done

@bertmert i'm not sure, someone can confirm that?

**Comment by yild** [comment]
.

**Comment by bertmert** [comment]
@yild 
Code is easier to read/understand if you use upper-case words for SQL statements like SELECT, DROP, ORDER BY, INTERVAL, MINUTE and so on. That's why I suggested to change line
```
$query->where('(a.modified > (a.created + interval ' . $this->getState('filter.modified_later_than', 0) . ' minute))');
```
like this:
```
$query->where('(a.modified > (a.created + INTERVAL ' . $this->getState('filter.modified_later_than', 0) . ' MINUTE))');
```
Thus it's clearer that `interval` or `minute` are not table fields/columns but SQL statements.

**Comment by yild** [comment]
@bertmert kk, I get your point

**Comment by brianteeman** [comment]
I am not convinced that this is something needed in the core - it seems to be an edge case<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14863">issues.joomla.org/tracker/joomla-cms/14863</a>.</sub>

**Comment by wojsmol** [comment]
@mbabker What do you think about this, are there chances to add this in Joomla 3.8.x?

**Comment by mbabker** [comment]
Personally agree with Brian, seems like an edge case scenario.  Personal feelings aside, without anyone indicating they've tested this, it's not going anywhere.

**Comment by wojsmol** [comment]
@mbabker With 2 successful tests until Friday would be a chance?

**Comment by mbabker** [comment]
That'll help, but at the end of the day if we evaluate the feature and decide it's not something that really fits into core, we won't merge it.

Plus, this will need testing/review against non-MySQL databases.

**Comment by brianteeman** [comment]
For me this is an edge case that should be solved with a custom module (you could easily fork the core module and install that)

**Comment by wojsmol** [comment]
I will test it tonight
cc @csthomas for non-mysql testing.

**Comment by wojsmol** [comment]
@brianteeman This add 8 line of php code and 1 parameter in xml. More changes will require the creation of a fork than a change in the core.

**Comment by brianteeman** [comment]
Maybe but speaking personally if something is to go into core then it should be something that many people will use. We can't put everything in the core just because there is code. We have far too many options already. More options makes Joomla harder to use for everyone. Plus don't forget it's not just 8 lines of code. There is documentation and ongoing support to consider as well.

**Comment by mbabker** [comment]
It's not about the number of lines of code being added, or even what it does.  Is it a functionality that there is a true need to support in core for a measurable number of our users?  There's no argument that it's not a nice to have thing if you need it, but it doesn't come across as something that a large number of users are looking for.

**Comment by wojsmol** [comment]
@mbabker @brianteeman I contacted @yild should speak out soon, he may convince you. I can do as much as I can, that is to test and try to get a second tester 😄

**Comment by brianteeman** [comment]
I am closing this for the reasons stated by myself and @mbabker above.
